qt-bugs@ issue : none
Trolltech task ID : 204403
bugs.kde.org number : none
applied: no
author: Mikko C. <mikko.cal@gmail.com>

Calling QSortFilterProxyModel::invalidate causes view
to scroll to the current item/index.
This behavior was not present on the Qt4.3.x. Fixed in Qt4.4.1 snapshots

--- a/src/gui/itemviews/qabstractitemview.cpp
+++ b/src/gui/itemviews/qabstractitemview.cpp
@@ -2980,7 +2980,7 @@
     if (q->isVisible()) {
         q->doItemsLayout();
         const QModelIndex current = q->currentIndex();
-        if (current.isValid())
+        if (current.isValid() && state == QAbstractItemView::EditingState)
             q->scrollTo(current);
     } else {
         doDelayedItemsLayout();
